src/react/reconcilation.js

```diff
@@ -80,26 +96,37 @@ export class Reconciler {
             // FatalError, unless it's a functional component that has no paramater
             // i.e let MyComponent = () => <div>Hello world</div>
             try {
-              let initialProps = getInitialProps(this.realm, componentType);
-              let initialContext = getInitialContext(this.realm, componentType);
+              let initialProps = props || getInitialProps(this.realm, componentType);
+              let initialContext = context || getInitialContext(this.realm, componentType);
               let { result } = this._renderComponent(componentType, initialProps, initialContext, "ROOT", null);
               this.statistics.optimizedTrees++;
               return result;
             } catch (error) {
+              // if we get an error and we're not dealing with the root
+              // rather than throw a FatalError, we log the error as a warning
+              // and continue with the other tree roots
+              // TODO: maybe control what levels gets treated as warning/error?
+              if (!isRoot) {
+                this.logger.logWarning(
+                  componentType,
+                  `__registerReactComponentRoot() React component tree (branch) failed due to - ${error.message}`
+                );
+                return this.realm.intrinsics.undefined;
```

Comment:
How does this `undefined` get used?

I understand it's a bail out but it's not clear to me what's going to happen with that `undefined` value. Is it just ignored somewhere?

Can you make a test for this case?

Reply:
Well, the reason this is used is because we need to return `Value` for effects. React component renders can't ever return `undefined` though as the reconciler blocks this, so if this occurs we know it's been done as part of a bail out.